### PR TITLE
Most popular items

### PR DESCRIPTION
**Pull request by jhennerich**
This branch adds a number of things.
 
added has_many relationships to many of our app/models.
implemented the faker and factory gems to the project 
created the most_popular_items method to merchant.rb and it's spec test

### CODE REVIEW

File: app/models/customer.rb [hunk 1]
```diff
@@ -4,6 +4,9 @@ class Customer < ApplicationRecord
 
   has_many :invoices
   has_many :transactions, through: :invoices
+  has_many :merchants, through: :invoices
```
Comment (Josenecal) [inline app/models/customer.rb:7]: After meeting with Dani, do we want to take these out unless we do end up needing them?

**Review by Josenecal — APPROVED** [review]
Overall this looks ready to merge! I guess it sounds like Dani would prefer us to not have relations that we're not using, but ostensibly I don't mind having them there at all (I'm not that worried about security consequences if we violate least privileged access here)

**Comment by jhennerich** [comment]
Per Joseph D request I am closing and splitting into 3 requests

**Comment by jhennerich** [comment]
I have spent over an hour trying to split this pull request into parts per Joesph's request and it is not working.  Joe  was able to pull this branch down and it was working for him.  I am reopening this request and am happy to step through each change with the team.

File: app/models/customer.rb (hunk 1 above)
Comment (jhennerich) [reply app/models/customer.rb:7]: I feel that I can explain why we will need these relationships in our model

**Review by josephdowns — APPROVED** [review]

**Pull request merged**